### PR TITLE
[Dynamic Instrumentation] Hotfix SymDB and ER

### PR DESCRIPTION
**Pull request by dudikeleti**
- Disable compresison in SymDB and fix out of range exception while collecting arguments of some methods
- Exception Replay should work even Live Debugger has not initialized

### CODE REVIEW

**Review by GreenMatan — APPROVED** [review]
LGTM

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (6468)](https://github.com/DataDog/dd-trace-dotnet/tree/77976dc244a861abe522ad34c7490d48b604cd59)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/aaaf96368f9fa425ac46483cb1a079092d72e6a4)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6468) - mean (68ms)  : 66, 71
     .   : milestone, 68,
    master - mean (69ms)  : 65, 73
     .   : milestone, 69,

    section CallTarget+Inlining+NGEN
    This PR (6468) - mean (975ms)  : 950, 1001
     .   : milestone, 975,
    master - mean (981ms)  : 957, 1005
     .   : milestone, 981,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6468) - mean (107ms)  : 105, 110
     .   : milestone, 107,
    master - mean (108ms)  : 105, 110
     .   : milestone, 108,

    section CallTarget+Inlining+NGEN
    This PR (6468) - mean (676ms)  : 657, 694
     .   : milestone, 676,
    master - mean (679ms)  : 666, 693
     .   : milestone, 679,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6468) - mean (91ms)  : 89, 93
     .   : milestone, 91,
    master - mean (91ms)  : 89, 93
     .   : milestone, 91,

    section CallTarget+Inlining+NGEN
    This PR (6468) - mean (635ms)  : 618, 652
     .   : milestone, 635,
    master - mean (637ms)  : 622, 651
     .   : milestone, 637,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6468) - mean (193ms)  : 190, 197
     .   : milestone, 193,
    master - mean (194ms)  : 189, 199
     .   : milestone, 194,

    section CallTarget+Inlining+NGEN
    This PR (6468) - mean (1,102ms)  : 1072, 1132
     .   : milestone, 1102,
    master - mean (1,100ms)  : 1064, 1136
     .   : milestone, 1100,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6468) - mean (278ms)  : 273, 283
     .   : milestone, 278,
    master - mean (278ms)  : 274, 282
     .   : milestone, 278,

    section CallTarget+Inlining+NGEN
    This PR (6468) - mean (873ms)  : 840, 905
     .   : milestone, 873,
    master - mean (870ms)  : 836, 905
     .   : milestone, 870,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6468) - mean (267ms)  : 262, 271
     .   : milestone, 267,
    master - mean (266ms)  : 262, 269
     .   : milestone, 266,

    section CallTarget+Inlining+NGEN
    This PR (6468) - mean (852ms)  : 819, 885
     .   : milestone, 852,
    master - mean (853ms)  : 825, 881
     .   : milestone, 853,

```

File: tracer/src/Datadog.Trace/Debugger/Upload/SymbolUploadApi.cs [hunk 1]
```diff
@@ -38,7 +38,7 @@ private SymbolUploadApi(
         {
             _apiRequestFactory = apiRequestFactory;
             _eventMetadata = eventMetadata;
-            _enableCompression = enableCompression;
+            _enableCompression = false;
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace/Debugger/Upload/SymbolUploadApi.cs:41]: nit: It seems strange to force this to `false` _here_ 😕 why not force it in the settings? At least a comment explaining why would probably be useful 😅

**Review by andrewlock — APPROVED** [review]
I'm not really sure where the bug was - in general it would be nice if there was a test showing the failure which is then fixed 🙂

File: tracer/src/Datadog.Trace/Debugger/Upload/SymbolUploadApi.cs (hunk 1 above)
Comment (dudikeleti) [reply tracer/src/Datadog.Trace/Debugger/Upload/SymbolUploadApi.cs:41]: I did it also in the settings but I'm too paranoid now 😅
I can revert it if it's too much.

**Comment by dudikeleti** [comment]
> I'm not really sure where the bug was - in general it would be nice if there was a test showing the failure which is then fixed 🙂

The issue was with the compressed symbols. It cause to missing symbols in all .net services.
There is a test in the debugger settings that makes sure  compression is disabled.

**Comment by andrewlock** [comment]
## Benchmarks Report for tracer :snail:

Benchmarks for #6468 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2): 
* 2 benchmarks are **slower**, with geometric mean 1.149
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.ActivityBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`StartStopWithChild`|net6.0|7.96μs|44.9ns|321ns|0.0119| 0.00397|0|5.62&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`StartStopWithChild`|netcoreapp3.1|10.2μs|56.3ns|356ns|0.0204| 0.0102|0|5.8&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`StartStopWithChild`|net472|16.4μs|42.8ns|166ns|1.05| 0.314|0.0966|6.2&nbsp;KB|
|#6468|`StartStopWithChild`|net6.0|8.08μs|45.2ns|300ns|0.0207| 0.00828|0|5.61&nbsp;KB|
|#6468|`StartStopWithChild`|netcoreapp3.1|10.1μs|56.9ns|386ns|0.0149| 0.00992|0|5.8&nbsp;KB|
|#6468|`StartStopWithChild`|net472|16.3μs|55.7ns|216ns|1.07| 0.326|0.114|6.21&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`WriteAndFlushEnrichedTraces`|net6.0|481μs|450ns|1.62μs|0| 0|0|2.7&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|640μs|296ns|1.15μs|0| 0|0|2.7&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`WriteAndFlushEnrichedTraces`|net472|847μs|887ns|3.32μs|0.425| 0|0|3.3&nbsp;KB|
|#6468|`WriteAndFlushEnrichedTraces`|net6.0|497μs|539ns|2.09μs|0| 0|0|2.7&nbsp;KB|
|#6468|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|650μs|718ns|2.78μs|0| 0|0|2.7&nbsp;KB|
|#6468|`WriteAndFlushEnrichedTraces`|net472|843μs|673ns|2.52μs|0.422| 0|0|3.3&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`SendRequest`|net6.0|156μs|893ns|6.8μs|0.143| 0|0|14.47&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`SendRequest`|netcoreapp3.1|180μs|1.24μs|12.3μs|0.171| 0|0|17.27&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`SendRequest`|net472|0.00296ns|0.00117ns|0.00453ns|0| 0|0|0&nbsp;b|
|#6468|`SendRequest`|net6.0|154μs|934ns|9.15μs|0.162| 0|0|14.47&nbsp;KB|
|#6468|`SendRequest`|netcoreapp3.1|168μs|981ns|8.88μs|0.182| 0|0|17.27&nbsp;KB|
|#6468|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
</details>


<details><summary><strong>Benchmarks.Trace.CIVisibilityProtocolWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`WriteAndFlushEnrichedTraces`|net6.0|565μs|3.1μs|19.1μs|0.296| 0|0|41.66&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|707μs|4.14μs|39.3μs|0.331| 0|0|41.79&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`WriteAndFlushEnrichedTraces`|net472|835μs|4.16μs|17.7μs|8.75| 2.5|0.417|53.27&nbsp;KB|
|#6468|`WriteAndFlushEnrichedTraces`|net6.0|559μs|3.11μs|20.1μs|0.571| 0|0|41.65&nbsp;KB|
|#6468|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|686μs|3.63μs|18.5μs|0.326| 0|0|41.9&nbsp;KB|
|#6468|`WriteAndFlushEnrichedTraces`|net472|803μs|1.98μs|7.14μs|8.41| 2.4|0.401|53.36&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`ExecuteNonQuery`|net6.0|1.27μs|1.79ns|6.94ns|0.0142| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`ExecuteNonQuery`|netcoreapp3.1|1.79μs|1.89ns|7.08ns|0.0135| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`ExecuteNonQuery`|net472|2.19μs|3.96ns|15.3ns|0.156| 0.00109|0|987&nbsp;B|
|#6468|`ExecuteNonQuery`|net6.0|1.28μs|1.84ns|7.14ns|0.0141| 0|0|1.02&nbsp;KB|
|#6468|`ExecuteNonQuery`|netcoreapp3.1|1.75μs|2.56ns|9.9ns|0.0139| 0|0|1.02&nbsp;KB|
|#6468|`ExecuteNonQuery`|net472|2.17μs|2.68ns|10.4ns|0.157| 0.00108|0|987&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`CallElasticsearch`|net6.0|1.22μs|0.52ns|1.95ns|0.0135| 0|0|976&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`CallElasticsearch`|netcoreapp3.1|1.53μs|0.756ns|2.83ns|0.013| 0|0|976&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`CallElasticsearch`|net472|2.53μs|0.959ns|3.59ns|0.157| 0|0|995&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`CallElasticsearchAsync`|net6.0|1.35μs|0.886ns|3.43ns|0.0134| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`CallElasticsearchAsync`|netcoreapp3.1|1.63μs|3.03ns|11.3ns|0.0138| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`CallElasticsearchAsync`|net472|2.64μs|0.953ns|3.44ns|0.167| 0|0|1.05&nbsp;KB|
|#6468|`CallElasticsearch`|net6.0|1.24μs|0.572ns|2.14ns|0.0137| 0|0|976&nbsp;B|
|#6468|`CallElasticsearch`|netcoreapp3.1|1.49μs|0.769ns|2.98ns|0.0128| 0|0|976&nbsp;B|
|#6468|`CallElasticsearch`|net472|2.51μs|0.887ns|3.32ns|0.157| 0|0|995&nbsp;B|
|#6468|`CallElasticsearchAsync`|net6.0|1.35μs|0.496ns|1.79ns|0.0136| 0|0|952&nbsp;B|
|#6468|`CallElasticsearchAsync`|netcoreapp3.1|1.68μs|1.09ns|4.21ns|0.0137| 0|0|1.02&nbsp;KB|
|#6468|`CallElasticsearchAsync`|net472|2.55μs|1.23ns|4.75ns|0.167| 0|0|1.05&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`ExecuteAsync`|net6.0|1.4μs|0.816ns|3.16ns|0.0133| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`ExecuteAsync`|netcoreapp3.1|1.69μs|0.585ns|2.27ns|0.0126| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`ExecuteAsync`|net472|1.83μs|0.365ns|1.37ns|0.145| 0|0|915&nbsp;B|
|#6468|`ExecuteAsync`|net6.0|1.43μs|0.613ns|2.37ns|0.0136| 0|0|952&nbsp;B|
|#6468|`ExecuteAsync`|netcoreapp3.1|1.67μs|0.457ns|1.71ns|0.0126| 0|0|952&nbsp;B|
|#6468|`ExecuteAsync`|net472|1.83μs|0.376ns|1.41ns|0.145| 0|0|915&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`SendAsync`|net6.0|4.51μs|1.11ns|4ns|0.0315| 0|0|2.31&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`SendAsync`|netcoreapp3.1|5.26μs|3.27ns|12.7ns|0.0368| 0|0|2.85&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`SendAsync`|net472|7.52μs|1.93ns|7.48ns|0.496| 0|0|3.12&nbsp;KB|
|#6468|`SendAsync`|net6.0|4.43μs|1.37ns|5.14ns|0.0308| 0|0|2.31&nbsp;KB|
|#6468|`SendAsync`|netcoreapp3.1|5.25μs|2.5ns|9.68ns|0.0365| 0|0|2.85&nbsp;KB|
|#6468|`SendAsync`|net472|7.28μs|1.1ns|4.11ns|0.495| 0|0|3.12&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`EnrichedLog`|net6.0|1.49μs|0.564ns|2.11ns|0.0232| 0|0|1.64&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`EnrichedLog`|netcoreapp3.1|2.27μs|0.733ns|2.74ns|0.0219| 0|0|1.64&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`EnrichedLog`|net472|2.68μs|1.1ns|4.1ns|0.249| 0|0|1.57&nbsp;KB|
|#6468|`EnrichedLog`|net6.0|1.48μs|0.826ns|3.09ns|0.023| 0|0|1.64&nbsp;KB|
|#6468|`EnrichedLog`|netcoreapp3.1|2.26μs|1.14ns|4.26ns|0.0226| 0|0|1.64&nbsp;KB|
|#6468|`EnrichedLog`|net472|2.69μs|1.32ns|5.11ns|0.249| 0|0|1.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`EnrichedLog`|net6.0|116μs|155ns|559ns|0| 0|0|4.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`EnrichedLog`|netcoreapp3.1|122μs|224ns|866ns|0.0608| 0|0|4.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`EnrichedLog`|net472|150μs|129ns|498ns|0.675| 0.225|0|4.46&nbsp;KB|
|#6468|`EnrichedLog`|net6.0|116μs|232ns|899ns|0.0581| 0|0|4.28&nbsp;KB|
|#6468|`EnrichedLog`|netcoreapp3.1|121μs|255ns|986ns|0| 0|0|4.28&nbsp;KB|
|#6468|`EnrichedLog`|net472|152μs|193ns|746ns|0.688| 0.229|0|4.46&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`EnrichedLog`|net6.0|3.07μs|1.14ns|4.41ns|0.0309| 0|0|2.2&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`EnrichedLog`|netcoreapp3.1|4.13μs|1.62ns|6.26ns|0.0292| 0|0|2.2&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`EnrichedLog`|net472|4.89μs|1.28ns|4.96ns|0.319| 0|0|2.02&nbsp;KB|
|#6468|`EnrichedLog`|net6.0|2.98μs|1.44ns|5.4ns|0.0303| 0|0|2.2&nbsp;KB|
|#6468|`EnrichedLog`|netcoreapp3.1|4.13μs|1.72ns|6.65ns|0.029| 0|0|2.2&nbsp;KB|
|#6468|`EnrichedLog`|net472|4.94μs|1.43ns|5.53ns|0.319| 0|0|2.02&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`SendReceive`|net6.0|1.41μs|1.39ns|5.2ns|0.0162| 0|0|1.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`SendReceive`|netcoreapp3.1|1.75μs|1.02ns|3.81ns|0.0157| 0|0|1.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`SendReceive`|net472|2.14μs|1.21ns|4.67ns|0.183| 0|0|1.16&nbsp;KB|
|#6468|`SendReceive`|net6.0|1.45μs|0.344ns|1.29ns|0.0159| 0|0|1.14&nbsp;KB|
|#6468|`SendReceive`|netcoreapp3.1|1.75μs|1.06ns|3.96ns|0.0149| 0|0|1.14&nbsp;KB|
|#6468|`SendReceive`|net472|2.12μs|2.91ns|11.3ns|0.183| 0|0|1.16&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`EnrichedLog`|net6.0|2.88μs|3ns|11.6ns|0.0216| 0|0|1.6&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`EnrichedLog`|netcoreapp3.1|3.85μs|3.69ns|14.3ns|0.0213| 0|0|1.65&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`EnrichedLog`|net472|4.19μs|2.82ns|10.9ns|0.323| 0|0|2.04&nbsp;KB|
|#6468|`EnrichedLog`|net6.0|2.85μs|1.19ns|4.47ns|0.0225| 0|0|1.6&nbsp;KB|
|#6468|`EnrichedLog`|netcoreapp3.1|3.88μs|1.92ns|7.44ns|0.0215| 0|0|1.65&nbsp;KB|
|#6468|`EnrichedLog`|net472|4.32μs|2.05ns|7.95ns|0.323| 0|0|2.04&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #6468

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.SpanBenchmark.StartFinishSpan&#8209;net6.0 | 1.172 | 397.50 | 465.95 | 
| Benchmarks.Trace.SpanBenchmark.StartFinishScope&#8209;netcoreapp3.1 | 1.126 | 684.80 | 771.05 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`StartFinishSpan`|net6.0|397ns|0.239ns|0.895ns|0.008| 0|0|576&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`StartFinishSpan`|netcoreapp3.1|617ns|0.43ns|1.67ns|0.00776| 0|0|576&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`StartFinishSpan`|net472|704ns|0.203ns|0.785ns|0.0918| 0|0|578&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`StartFinishScope`|net6.0|558ns|0.432ns|1.67ns|0.0098| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`StartFinishScope`|netcoreapp3.1|685ns|0.368ns|1.42ns|0.00942| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`StartFinishScope`|net472|897ns|0.523ns|2.03ns|0.104| 0|0|658&nbsp;B|
|#6468|`StartFinishSpan`|net6.0|466ns|0.413ns|1.6ns|0.00811| 0|0|576&nbsp;B|
|#6468|`StartFinishSpan`|netcoreapp3.1|557ns|0.246ns|0.92ns|0.00783| 0|0|576&nbsp;B|
|#6468|`StartFinishSpan`|net472|687ns|0.349ns|1.35ns|0.0915| 0|0|578&nbsp;B|
|#6468|`StartFinishScope`|net6.0|570ns|0.237ns|0.919ns|0.00976| 0|0|696&nbsp;B|
|#6468|`StartFinishScope`|netcoreapp3.1|772ns|0.709ns|2.65ns|0.00931| 0|0|696&nbsp;B|
|#6468|`StartFinishScope`|net472|827ns|0.626ns|2.26ns|0.105| 0|0|658&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`RunOnMethodBegin`|net6.0|615ns|0.399ns|1.54ns|0.00958| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`RunOnMethodBegin`|netcoreapp3.1|1μs|0.866ns|3.35ns|0.00899| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/df53d901c4d8136f8017a8490148b00ae3acbeb2)|`RunOnMethodBegin`|net472|1.14μs|1.08ns|4.19ns|0.104| 0|0|658&nbsp;B|
|#6468|`RunOnMethodBegin`|net6.0|610ns|0.168ns|0.649ns|0.00973| 0|0|696&nbsp;B|
|#6468|`RunOnMethodBegin`|netcoreapp3.1|964ns|0.631ns|2.36ns|0.00917| 0|0|696&nbsp;B|
|#6468|`RunOnMethodBegin`|net472|1.13μs|0.662ns|2.56ns|0.104| 0|0|658&nbsp;B|
</details>

**Pull request merged**